File: Source/Obsequy.Web/App/Forms/passwordRecoveryForm.js


angular.module('passwordRecovery-form', [])
	.factory('PasswordRecoveryForm', [function () {
		var PasswordRecoveryForm = function (properties) {

 			angular.extend(this, {

 				email: undefined,

 				securityQuestion: undefined,
 				securityAnswer: undefined,

 				canAnswerSecurityQuestion: undefined,
 				canResetPassword: undefined,
 				canSendSmSCode: undefined,

 				isAnsweringSecurityQuestion: undefined,
 				isResettingPassword: undefined,
 				isSendingSmSCode: undefined,

 				update: function (properties) {
 					properties = properties || {};

					// update properties
 					this.email = properties.email;
 				
 					this.securityQuestion = properties.securityQuestion;
 					this.securityAnswer = properties.securityAnswer;

 					this.canAnswerSecurityQuestion = properties.canAnswerSecurityQuestion;
 					this.canResetPassword = properties.canResetPassword;
 					this.canSendSmSCode = properties.canSendSmSCode;

 					this.isAnsweringSecurityQuestion = properties.isAnsweringSecurityQuestion;
 					this.isResettingPassword = properties.isResettingPassword;
 					this.isSendingSmSCode = properties.isSendingSmSCode;
				},

				toJSON: function () {
					return {
						email: this.email,
						securityQuestion: this.securityQuestion,
						securityAnswer: this.securityAnswer,
						canAnswerSecurityQuestion: this.canAnswerSecurityQuestion,
						canResetPassword: this.canResetPassword,
						canSendSmSCode: this.canSendSmSCode,
						isAnsweringSecurityQuestion: this.isAnsweringSecurityQuestion,
						isResettingPassword: this.isResettingPassword,
						isSendingSmSCode: this.isSendingSmSCode
					};
				}
			});

			this.update(properties);
		};

		return PasswordRecoveryForm;
	}]);

File: Source/Obsequy.Model/Form/PasswordRecoveryForm.cs
using FluentValidation.Results;
using Obsequy.Utility;

namespace Obsequy.Model
{
	public class PasswordRecoveryForm
	{
		#region Properties

		public string Email { get; set; }

		public string SecurityQuestion { get; set; }
		public string SecurityAnswer { get; set; }

		public bool CanAnswerSecurityQuestion { get; set; }
		public bool CanResetPassword { get; set; }
		public bool CanSendSmSCode { get; set; }

		public bool IsAnsweringSecurityQuestion { get; set; } // they've elected to answer the security question
		public bool IsResettingPassword { get; set; }  // they just want a reset link to follow
		public bool IsSendingSmSCode { get; set; }    // they want us to send them an SMS code

		#endregion

		public PasswordRecoveryForm()
		{
		}

		public ValidationResult Validate()
		{
			return ((new PasswordRecoveryFormValidator()).Validate(this) as ValidationResult);
		}
	}
}
